pkg/jobs/job_scheduler.go

```diff
@@ -360,14 +370,27 @@ func (s *jobScheduler) runDaemon(ctx context.Context, stopper *stop.Stopper) {
 			log.Errorf(ctx, "error registering executor metrics: %+v", err)
 		}
 
+		schedulerEnabledOnThisNode := func() bool {
+			if s.ShouldRunScheduler == nil || !schedulerRunsOnSingleNode.Get(&s.Settings.SV) {
+				return true
+			}
+
+			enabled, err := s.ShouldRunScheduler(ctx, s.DB.Clock().NowAsClockTimestamp())
+			if err != nil {
+				log.Errorf(ctx, "error determining if the scheduler enabled: %v; will recheck after %s",
+					err, recheckEnabledAfterPeriod)
+				return false
+			}
+			return enabled
+		}
+
 		for timer := time.NewTimer(initialDelay); ; timer.Reset(
```

Comment:
Huh, I know this is existing code, but I'm surprised the linter allowed time.NewTimer and didn't say to use timeutil here, and then the usual timer.Read = true below I the <- case.

Reply:
Want to change that?

Reply:
no, follow up.